### DISCUSSION TITLE
Show call stack of running thread in debugger extension

### DISCUSSION THREAD

**Discussion by felixhirt** [microsoft/vscode-discussions#753] [Extension Development QnA] [answered]
I am developing a debugger extension for a language/runtime which can show the call stack for running threads (or rather a snapshot at the moment of requesting), not just stopped ones. Is there a way to get vs code to periodically request and display the call stack of a running thread through "stackTraceRequest" (either directly in the debugger or by sending a specific event to vs code)?

**Comment by isidorn** [comment, marked as answer]
I am not 100% sure if this is possible. VS Code might assume that only stopped threads have call stacks.
I suggest to just try it out and see if it works. If it does not, you can file a new issue in the VS Code repo. Since I think the DAP spec does not prohibit this.

**Comment by felixhirt** [comment]
Issue: https://github.com/microsoft/vscode/issues/190554
